### PR TITLE
Merge mlsld into master

### PR DESCRIPTION
**Pull request by kazbaig**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/web-ui/src/public/Main.vue [hunk 1]
```diff
@@ -1,249 +1,104 @@
...
-          else {
-            // Assume popular products are being displayed until user generates some events
-            this.display = 'Trending products';
-          }
-          this.personalized = true
-          this.explain_recommended = 'Personalize recipe: ' + response.headers['x-personalize-recipe']
-        }
-        if (response.headers['x-experiment-name']) {
-          // Can't be sure if we're personalizing or not and if so to what degree
-          this.display = 'Recommended for you';
```
Comment (james-jory) [inline src/web-ui/src/public/Main.vue:196]: It doesn't appear that the new cold user section header behavior was ported correctly to the new UX. There are now 4 possible display headings: "Featured", "Trending products", "Inspired by your shopping trends", and "Recommended for you". Looks like the PR has us back to the old "Featured" and "Inspired by your shopping trends".

File: src/web-ui/src/public/ProductDetail.vue [hunk 2]
```diff
@@ -1,233 +1,247 @@
...
-    <div class="col-sm-12 col-md-6 col-lg-6">
-       <h5>{{ product.name }}</h5>
-       <p>{{ product.description }}</p>
-       <p>${{ product.price }}</p>
-       <span v-if="product.current_stock > 0">Items currently in stock: {{ product.current_stock }}</span>
-       <span v-else>Sorry, this item is currently out of stock</span>
-       <p>
-        <i class="fas fa-star"></i><i class="fas fa-star"></i><i class="fas fa-star"></i><i class="fas fa-star"></i><i class="fas fa-star"></i>
-       </p>       
-       <button class="btn btn-outline-primary" v-on:click="addToCart()" :disabled='addToCartDisabled'> Add to Cart </button>
```
Comment (james-jory) [inline src/web-ui/src/public/ProductDetail.vue:23]: Looks like the PR has lost the new product stock behavior.

**Review by james-jory — CHANGES_REQUESTED** [review]
Looks good. A couple comments that appear to be regressions that were missed in the merge.

File: src/web-ui/src/public/Main.vue (hunk 1 above)
Comment (kazbaig) [reply src/web-ui/src/public/Main.vue]: Updated to dynamically display 1 of the 3 possible headings within `getUserRecommendations` now, in addition to the featured products section regardless of authentication.

File: src/web-ui/src/public/ProductDetail.vue (hunk 2 above)
Comment (kazbaig) [reply src/web-ui/src/public/ProductDetail.vue:23]: Added this back with some additional rules around when to display that quantity dropdown and with what available quantities.

File: src/web-ui/src/public/Main.vue [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <Layout>
     <div class="content container">
       <RecommendedProductsSection
-        v-if="personalizeUserID"
+        v-if="user && personalizeUserID"
```
Comment (james-jory) [inline src/web-ui/src/public/Main.vue:5]: The new cold user behavior does not require a user, only a `personalizeUserID`. Remove the `user` check here.

File: src/web-ui/src/public/ProductDetail.vue [hunk 4]
```diff
@@ -97,6 +114,19 @@ export default {
         text: this.readableProductCategory,
       };
     },
+    cartItem() {
+      if (!this.product || !this.cart) return 0;
```
Comment (james-jory) [inline src/web-ui/src/public/ProductDetail.vue:118]: Should `null` be returned here instead?

**Review by james-jory — CHANGES_REQUESTED** [review]
The logic in Main.vue is still not right. Please review the comments and changes in Main.vue from the PR.

https://github.com/aws-samples/retail-demo-store/pull/116/files#diff-3a06c30fc47f54a423669be3a0377802be1aa884a154fecaecac6db9cfc83680R178

A user is no longer required for personalized recommendations, only a `personalizeUserID`. Whether recommendations are personalized or not is captured by the `personalizeRecommendationsForVisitor` state. This should not be used as gating logic at the top of `getUserRecommendations()`.

**Review by alexchirayath — APPROVED** [review]
LGTM

File: src/web-ui/src/public/ProductDetail.vue (hunk 4 above)
Comment (alexchirayath) [reply src/web-ui/src/public/ProductDetail.vue]: Good point. Its updated in the #122

File: src/web-ui/src/public/Main.vue (hunk 3 above)
Comment (alexchirayath) [reply src/web-ui/src/public/Main.vue]: Updated in #122

**Review by james-jory — APPROVED** [review]

**Pull request merged**